### PR TITLE
[sysrst_ctrl,sival] Port sysrst_ctrl_outputs

### LINKED ISSUE #21066: [chip-test] chip_sw_sysrst_ctrl_outputs
### Test point name

[chip_sw_sysrst_ctrl_outputs](https://github.com/lowRISC/opentitan/blob/master/hw/top_earlgrey/data/ip/chip_sysrst_ctrl_testplan.hjson#L48)

### Host side component

Rust

### OpenTitanTool infrastructure implemented

Yes

### Silicon Validation (SiVal)

Yes

### Emulation Targets

- [ ] None
- [ ] CW310
- [X] Hyperdebug + CW310

### Contact person

@rswarbrick 

### Checklist

Please fill out this checklist as items are completed. Link to PRs and issues as appropriate.
- [x] Check if existing test covers most or all of this testpoint (if so, either extend said test to cover all points, or skip the next 3 checkboxes)
- [ ] Device-side (C) component developed
- [ ] Bazel build rules developed
- [ ] Host-side component developed
- [x] HJSON test plan updated with test name (so it shows up in the dashboard)
- [ ] Test added to dvsim nightly regression (and passing at time of checking)
- [ ] For SiVal test cases, test is running relevant FPGA or silicon regression

> estimate 1

### PR DESCRIPTION
**Pull request by pamaury**
Fixes #21066

### CODE REVIEW

File: sw/host/tests/chip/sysrst_ctrl/sysrst_ctrl_ulp_z3_wakeup.rs [hunk 1]
```diff
@@ -57,6 +57,8 @@ static CONFIG: Lazy<HashMap<&'static str, Config>> = Lazy::new(|| {
             open_drain: vec![false, false, false],
             // z3_wakeup
             input_pins: vec!["IOR5"],
+            // ec_rst, flash_wp
```
Comment (jwnrt) [inline sw/host/tests/chip/sysrst_ctrl/sysrst_ctrl_ulp_z3_wakeup.rs:60]: Nit: none set

File: sw/device/tests/sysrst_ctrl_outputs_test.c [hunk 2]
```diff
@@ -77,16 +94,28 @@ static const dif_sysrst_ctrl_pin_t kSysrstCtrlOutputs[] = {
...
+OT_SECTION(".rodata")
+static volatile const uint8_t kTestPhaseDV = 0;
+OT_SECTION(".data")
+static volatile const uint8_t kTestPhaseReal = kTestPhaseSetup;
 
 // Sets up the pinmux to assign input and output pads
 // to the sysrst_ctrl peripheral as required.
 static void pinmux_setup(void) {
+  /* On real devices, we also need to configure the DIO pins */
+  if (kDeviceType != kDeviceSimDV)
```
Comment (jwnrt) [inline sw/device/tests/sysrst_ctrl_outputs_test.c:109]: Nit: I think our style requires braces for `if` statements

File: sw/host/tests/chip/sysrst_ctrl/sysrst_ctrl_outputs.rs [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    Done = 7,
+}
+
+static CONFIG: Lazy<HashMap<&'static str, Config>> = Lazy::new(|| {
+    collection! {
+        "hyper310" => Config {
+            // key0_in, key1_in, key2_in, pwrb_in
+            output_pins: vec!["IOR10", "IOR11", "IOR12", "IOR5", ],
+            open_drain: vec![false, false, false, false],
+            // key0_out, key1_out, key2_out, pwrb_out, bat_dis_out, z3_wakeup_out, ec_rst, flash_wp
```
Comment (jwnrt) [inline sw/host/tests/chip/sysrst_ctrl/sysrst_ctrl_outputs.rs:65]: I wonder if we should use aliases for these pins so we can refer to them directly by name. Don't worry about it for this PR, though. I know the alias-mapping code can get quite spaghetti.

File: sw/host/tests/chip/sysrst_ctrl/sysrst_ctrl_outputs.rs [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+fn read_loopback_pads(params: &Params) -> Result<u8> {
+    // Only capture the pins of interest.
+    Ok(read_pins(params.transport, params.config)? & LOOPBACK_ALL_SET)
+}
+
+fn read_output_pads(params: &Params) -> Result<u8> {
+    read_pins(params.transport, params.config)
+}
+
+fn check_loopback_single(params: &Params) -> Result<()> {
```
Comment (jwnrt) [inline sw/host/tests/chip/sysrst_ctrl/sysrst_ctrl_outputs.rs:92]: Nit: I was slightly confused that the `check_...` functions also set the loopback, but not a huge issue. I'm not sure I have a constructive suggestion for a different name though, sorry!

**Review by jwnrt — APPROVED** [review]
The code here looks good, thanks @pamaury. Only nits.

File: sw/host/tests/chip/sysrst_ctrl/sysrst_ctrl_outputs.rs (hunk 3 above)
Comment (pamaury) [reply sw/host/tests/chip/sysrst_ctrl/sysrst_ctrl_outputs.rs:65]: Except for ec_rst and flash_wp, those MIOS that can be routed to any pin though so I am not sure an alias is ideal. But I think applying your suggestion to those two pins makes sense though! I'll create a separate PR for that.

File: sw/host/tests/chip/sysrst_ctrl/sysrst_ctrl_outputs.rs (hunk 4 above)
Comment (pamaury) [reply sw/host/tests/chip/sysrst_ctrl/sysrst_ctrl_outputs.rs:92]: I followed closely the DV test. Would it help if I rename it to something like `check_loopback_works`? With a verb, it's clearer that it is not just reading?

File: sw/host/tests/chip/sysrst_ctrl/sysrst_ctrl_ulp_z3_wakeup.rs (hunk 1 above)
Comment (pamaury) [reply sw/host/tests/chip/sysrst_ctrl/sysrst_ctrl_ulp_z3_wakeup.rs:60]: oops, thank you!

**Comment by pamaury** [comment]
Thanks for the comments @jwnrt , I have addressed several.

File: sw/host/tests/chip/sysrst_ctrl/sysrst_ctrl_outputs.rs (hunk 4 above)
Comment (jwnrt) [reply sw/host/tests/chip/sysrst_ctrl/sysrst_ctrl_outputs.rs:92]: Hmm I think probably the current name is good enough, I won't bikeshed it too much

**Review by engdoreis — APPROVED** [review]

**Comment by pamaury** [comment]
Thanks @engdoreis , I have addressed your comment.

**Pull request merged**